gallery/gallery-library/src/main/java/org/silverpeas/components/gallery/Watermark.java

```diff
@@ -54,14 +74,70 @@ void setTextForThumbnails(final String textForThumbnails) {
     this.textForThumbnails = textForThumbnails;
   }
 
+  File getImageForHD() {
+    return imageForHD;
+  }
+
+  synchronized void setImageUrlForHD(String componentInstanceId, final String imageUrlForHD) {
+    this.imageForHD = processImageUrl(componentInstanceId, imageUrlForHD);
+  }
+
+  File getImageForThumbnails() {
+    return imageForThumbnails;
+  }
+
+  synchronized void setImageUrlForThumbnails(String componentInstanceId,
+      final String imageUrlForThumbnails) {
+    this.imageForThumbnails = processImageUrl(componentInstanceId, imageUrlForThumbnails);
+  }
+
+  private synchronized File processImageUrl(String componentInstanceId, final String imageUrl) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3776)